addons/api/addon/models/role.js

```diff
@@ -33,6 +42,66 @@ export default class RoleModel extends GeneratedRoleModel {
    */
   @attr({ readOnly: true, emptyArrayIfMissing: true }) grant_strings;
 
+  /**
+   * Convenience for looking up the grant scopes, if loaded.
+   */
+  get grantScopes() {
+    const grantScopes = [];
+    if (this.grant_scope_ids) {
+      const sortedScopeIDs = this.grant_scope_ids.slice().sort((a, b) => {
+        const aIndex = GRANT_SCOPE_KEYWORDS.indexOf(a);
+        const bIndex = GRANT_SCOPE_KEYWORDS.indexOf(b);
+
+        // both a and b are keywords
+        if (aIndex !== -1 && bIndex !== -1) {
+          return aIndex - bIndex;
+        }
+        // only a or b is a keyword
+        if (aIndex !== -1) return -1;
+        if (bIndex !== -1) return 1;
+
+        // only a is global
+        if (a === 'global') return -1;
+        // only b is global
+        if (b === 'global') return 1;
+
+        // both a and b are orgs - keep original order
+        if (a.startsWith('o_') && b.startsWith('o_')) return 0;
+        // only a is an org
+        if (a.startsWith('o_')) return -1;
+        // only b is an org
+        if (b.startsWith('o_')) return 1;
+
+        // both a and b are projects - keep original order
+        if (a.startsWith('p_') && b.startsWith('p_')) return 0;
+        // only a is a project
+        if (a.startsWith('p_')) return -1;
+        // only b is a project
+        if (b.startsWith('p_')) return 1;
+
+        // default - keep original order
+        return 0;
```

Comment:
I see what you mean by this being overkill, but I think the logic can be simplified a bit more by using your original sorting idea with the `GRANT_SCOPE_KEYWORDS`.

If you have an array or object like this:
```
{ 
  "this": 0,
  "children": 1,
  "descendants": 2,
  "global": 3,
  "o": 4,
  "p": 5
}
```

You can still do the sort function logic with `sortOrder.indexOf(a) - sortOrder.indexOf(b)` or `sortOrder[a] - sortOrder[b]` depending if you use an array or object (I don't think there's a big performance difference for iterating over such a small array).

The only gotcha is you have to do a little processing first if the ids are orgs or project ones since you have to slice off the extra characters after the underscore, but that's relatively straightforward since you could just do something like `a.split('_')[0]` applied to all your ids.

Reply:
I see what you are getting at. I will update the sort now.

Reply:
Updated. Thanks for the suggestion.